synapseclient/client.py

```diff
@@ -2098,15 +2098,21 @@ def getPermissions(self, entity, principalId=None):
                   or an empty array
 
         """
-        # TODO: what if user has permissions by membership in a group?
-        principalId = self._getUserbyPrincipalIdOrName(principalId)
+        principal_id = self._getUserbyPrincipalIdOrName(principalId)
         acl = self._getACL(entity)
+
+        team_list = self._find_teams_for_principal(principal_id)
```

Comment:
In the docstring I have:
```
        """
        Retrieve a list of teams for the matching principal ID. **If the principalId that is passed in is a team itself,
        or not found, this will return a generator that yields no results.**

        :param principal_id: Identifier of a user or group.

        :return:  A generator that yields objects of type :py:class:`synapseclient.team.Team`
        """
```

When I tested it the REST API returned an empty set of results.